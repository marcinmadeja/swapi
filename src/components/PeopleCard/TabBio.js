import React from 'react';

import {
  Content,
  Item,
  Label,
  Value,
} from './TabBio.styles';

const TabBio = ({
  userData: {
    birth_year,
    eye_color,
    hair_color,
    height,
    mass,
    skin_color,
  },
}) => {
  return (
    <Content>
      <Item>
        <Label>Birthday:</Label>
        <Value>{birth_year}</Value>
      </Item>

      <Item>
        <Label>Eye color:</Label>
        <Value>{eye_color}</Value>
      </Item>

      <Item>
        <Label>hair color:</Label>
        <Value>{hair_color}</Value>
      </Item>

      <Item>
        <Label>Height:</Label>
        <Value>{height}</Value>
      </Item>

      <Item>
        <Label>Mass:</Label>
        <Value>{mass}</Value>
      </Item>

      <Item>
        <Label>Skin color:</Label>
        <Value>{skin_color}</Value>
      </Item>
    </Content>
  );
};

export default TabBio;
