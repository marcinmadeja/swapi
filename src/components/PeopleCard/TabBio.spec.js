import React from 'react';
import 'jest-styled-components';
import TabBio from './TabBio';

const userData = {
  birth_year: '1999',
  eye_color: 'red',
  hair_color: 'red',
  height: '180',
  mass: '90',
  skin_color: 'red',
};

describe('TabBio', () => {
  it('should match snapshot', () => {
    const component = <TabBio userData={userData} />;
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });
});
