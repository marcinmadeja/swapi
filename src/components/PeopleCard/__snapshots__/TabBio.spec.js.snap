// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
.c0 {
  max-width: 300px;
  margin: 0 auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  padding: 0 10px;
  border-bottom: 1px solid #e6ecf5;
}

.c1:last-child {
  border-bottom: none;
}

.c2 {
  width: 150px;
}

.c3 {
  margin-left: auto;
  color: #607d8b;
  font-size: 18px;
  font-weight: 500;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Birthday:
    </div>
    <div
      className="c3"
    >
      1999
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Eye color:
    </div>
    <div
      className="c3"
    >
      red
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      hair color:
    </div>
    <div
      className="c3"
    >
      red
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Height:
    </div>
    <div
      className="c3"
    >
      180
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Mass:
    </div>
    <div
      className="c3"
    >
      90
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Skin color:
    </div>
    <div
      className="c3"
    >
      red
    </div>
  </div>
</div>
`;
